src/include/nodes/relation.h

```diff
@@ -322,11 +322,9 @@ typedef struct PlannerInfo
 typedef struct CtePlanInfo
 {
 	/*
-	 * A subplan, prepared for sharing among many CTE references by
-	 * prepare_plan_for_sharing(), that implements the CTE. NULL if the
-	 * CTE is not shared among references.
+	 * A subplan that implements the CTE.
```

Comment:
Well, according to new comment the user can misunderstand it and think that it's guaranteed that subplan here is always not NULL.
However, subplan will be set here only for the cases of plan sharing or for the case described by that patch. 
Therefore, the comment should be rewritten and should describe the true state of affairs.